File: Week_03/G20200343030005/Leetcode_455_001.js
/**
455. 分发饼干
*/


var findContentChildren = (g, s) => {
  g.sort((a,b)=>a-b)
  s.sort((a,b)=>a-b)
  let len = g.length
  while (s.length) {
    if (s.shift() >= g[0]) g.shift()
  }
  return len - g.length
};

File: Week_03/G20200343030001/Leetcode_455_001.java
package Week_03;

import java.util.Arrays;

public class Leetcode_455_001 {
    public int findContentChildren(int[] g, int[] s) {
        if(g == null || s == null) {
            return 0;
        };

        Arrays.sort(g);
        Arrays.sort(s);

        int gi = 0, si = 0;

        while(gi < g.length && si < s.length) {
            if(g[gi] <= s[si]) {
                gi++;
            }

            si++;
        }

        return gi;
    }
}
